### PR TITLE
Adds updated dependency check 

### PR DESCRIPTION
**Pull request by jom**
Removes old legacy WC classes and adds check for Sensei 2.x. It also bumps PHP version, but we could just lower it for now and bump it up when needed.

### CODE REVIEW

**Comment by jom** [comment]
@alexsanford Haven’t tested with just WooThemes Updater. Didn’t think it was being maintained. I’ll do that tomorrow.

Do you have ideas for trimming it down? I don’t see another way to do this. One thing I tried to do to minimize maintenance is make all methods except the main one private. Will hopefully help with future changes and minimize depreciations.

File: sensei-content-drip.php [hunk 1]
```diff
@@ -19,6 +19,11 @@
 	exit;
 }
 
+require_once __DIR__ . '/includes/class-sdc-ext-dependency-checker.php';
```
Comment (donnapep) [inline sensei-content-drip.php]: This should be "scd", not "sdc". 🙂

File: includes/class-sdc-ext-dependency-checker.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	public static function add_php_notice() {
+		$screen        = get_current_screen();
+		$valid_screens = array( 'dashboard', 'plugins' );
+
+		if ( ! current_user_can( 'activate_plugins' ) || ! in_array( $screen->id, $valid_screens, true ) ) {
+			return;
+		}
+
+		// translators: %1$s is version of PHP that SCD requires; %2$s is the version of PHP WordPress is running on.
+		$message = sprintf( __( '<strong>Sensei Content Drip</strong> requires PHP version %1$s but you are running %2$s.', 'sensei-content-drop' ), self::MINIMUM_PHP_VERSION, phpversion() );
```
Comment (donnapep) [inline includes/class-sdc-ext-dependency-checker.php]: drop => drip

File: includes/class-sdc-ext-dependency-checker.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	public static function add_sensei_notice() {
+		$screen        = get_current_screen();
+		$valid_screens = array( 'dashboard', 'plugins' );
+
+		if ( ! current_user_can( 'activate_plugins' ) || ! in_array( $screen->id, $valid_screens, true ) ) {
+			return;
+		}
+
+		// translators: %1$s is the minimum version number of Sensei that is required.
+		$message = sprintf( __( '<strong>Sensei Content Drip</strong> requires the plugin <strong>Sensei</strong> (minimum version: <strong>%1$s</strong>) to be installed and activated.', 'sensei-content-drop' ), self::MINIMUM_SENSEI_VERSION );
```
Comment (donnapep) [inline includes/class-sdc-ext-dependency-checker.php]: drop => drip

**Comment by donnapep** [comment]
Just curious where the 1.11.0 minimum version of Sensei came from?

Also, when I installed Sensei 1.11.0 I still got the Sensei notice. Same thing for Sensei 2.0.0-beta.1.

**Comment by jom** [comment]
@donnapep Make sure you're removing the options `woothemes-sensei-version` _and_ `sensei-version` when switching between versions and then deactivate/reactivate.

I thought supporting two majors back would be good. Didn't think we wanted to test further back than that, but could also see 1.12.0 being fine.

**Comment by donnapep** [comment]
> Make sure you're removing the options woothemes-sensei-version and sensei-version when switching between versions and then deactivate/reactivate Sensei.

It only seems to happen when Sensei is in a non-standard folder (e.g. `sensei-1.11.0`). Sometimes it finds `Sensei_Main` and sometimes it doesn't. Because I can't reliably reproduce it though there's nothing we really need to do, but just wanted to note it somewhere.

> I thought supporting two majors back would be good. Didn't think we wanted to test further back than that, but could also see 1.12.0 being fine.

Makes sense.

**Pull request merged**